### PR TITLE
Clean up MTRClusters/MTRBaseClusters a bit:

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
1) Introduce MTRAbstractBaseCluster/MTRAbstractCluster base classes. 
2) Move the same-for-every-type initializer to the base class. 
3) Move the class-specific initializer declaration (needed to get availability
   right) to a category so we don't get complaints about incomplete
   implementation.
4) Remove asByteSpan/asCharSpan instance methods on MTRCluster, just use the
   helpers directly in generated code.

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@bzbarsky-apple FYI

**Comment by bzbarsky-apple** [comment]
Yeah, I need to figure out whether we actually want this change.

**Comment by bzbarsky-apple** [comment]
@ksperling-apple suggested we do MTRGenericCluster and MTRGenericBaseCluster, which I like more than "abstract".  That would also let us eventually put any functionality that just needs a cluster id on those interfaces....

File: src/darwin/Framework/CHIP/MTRCluster.h [hunk 1]
```diff
@@ -46,6 +47,29 @@ NS_ASSUME_NONNULL_BEGIN
...
+- (instancetype)init NS_UNAVAILABLE;
++ (instancetype)new NS_UNAVAILABLE;
+@end
+
+/**
+ * Base for all MTRCluster* types.
+ */
+MTR_NEWLY_AVAILABLE
+@interface MTRGenericCluster : MTRCluster
+- (instancetype)init NS_UNAVAILABLE;
```
Comment (ksperling-apple) [inline src/darwin/Framework/CHIP/MTRCluster.h:64]: If init and new are already unavailable on MTRCluster you don’t need to repeat it here and below

**Review by ksperling-apple — APPROVED** [review]

File: src/darwin/Framework/CHIP/MTRCluster.h (hunk 1 above)
Comment (bzbarsky-apple) [reply src/darwin/Framework/CHIP/MTRCluster.h]: Done.

**Comment by bzbarsky-apple** [comment]
Fast-tracking platform-specific change with platform owner review.

**Pull request merged**